token-lending/sdk/src/state/obligation.rs

```diff
@@ -410,20 +473,91 @@ impl ObligationLiquidity {
 
         Ok(())
     }
+
+    /// Calculates shares for liquidity mining.
+    pub fn liability_shares(&self) -> Result<u64, ProgramError> {
+        self.borrowed_amount_wads
+            .try_div(self.cumulative_borrow_rate_wads)?
+            .try_floor_u64()
+    }
 }
 
 const OBLIGATION_COLLATERAL_LEN: usize = 88; // 32 + 8 + 16 + 32
 const OBLIGATION_LIQUIDITY_LEN: usize = 112; // 32 + 16 + 16 + 16 + 32
-const OBLIGATION_LEN: usize = 1300; // 1 + 8 + 1 + 32 + 32 + 16 + 16 + 16 + 16 + 64 + 1 + 1 + (88 * 1) + (112 * 9)
-                                    // @TODO: break this up by obligation / collateral / liquidity https://git.io/JOCca
-impl Pack for Obligation {
-    const LEN: usize = OBLIGATION_LEN;
+/// This is the size of the account _before_ LM feature was added.
+const OBLIGATION_LEN_V2_0_2: usize = 1300; // 1 + 8 + 1 + 32 + 32 + 16 + 16 + 16 + 16 + 64 + 1 + 1 + (88 * 1) + (112 * 9)
+                                           // @TODO: break this up by obligation / collateral / liquidity https://git.io/JOCca
+impl Obligation {
+    /// Obligation with no Liquidity Mining Rewards
+    pub const MIN_LEN: usize = OBLIGATION_LEN_V2_0_2;
+
+    /// Maximum account size for obligation.
+    /// Scenario in which all reserves have all associated rewards filled.
+    ///
+    /// - [Self::user_reward_managers] vec length in u8
+    /// - [Self::user_reward_managers] vector
+    pub const MAX_LEN: usize =
+        Self::MIN_LEN + 1 + MAX_OBLIGATION_RESERVES * UserRewardManager::MAX_LEN;
```

Comment:
```suggestion
    /// Maximum account size for obligation.
    /// Scenario in which all reserves have all associated rewards filled.
    ///
    /// - [Self::user_reward_managers] vec length in u8
    /// - [Self::user_reward_managers] vector per each reserve
    /// - Finally we multiply the number of reserves by two because both
    ///   deposit and borrow reserve can be in the obligation 
    pub const MAX_LEN: usize =
        Self::MIN_LEN + 1 + MAX_OBLIGATION_RESERVES * UserRewardManager::MAX_LEN * 2;
```

This is actually wrong because you can have borrow & lend reward for each position.

Reply:
No longer relevant with audit fixes